File: implementations/java/src/main/java/com/axisrobo/aep/Envelope.java
package com.axisrobo.aep;

import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

public final class Envelope {
    private Envelope() {}

    public static List<String> validate(Map<String, Object> value) {
        if (value == null || value.get("type") instanceof String type
            && com.axisrobo.harmovela.event.registry.EventTypes.isStandardEventType(type)) {
            return com.axisrobo.harmovela.event.envelope.Envelope.validate(value);
        }
        var adapted = new LinkedHashMap<>(value);
        var type = (String) value.get("type");
        adapted.put("type", "event.acknowledged");
        var errors = new java.util.ArrayList<>(com.axisrobo.harmovela.event.envelope.Envelope.validate(adapted));
        if (!EventTypes.isStandardEventType(type)) errors.add("type is not in the standard draft registry: " + type);
        return errors;
    }
}

File: implementations/go/aep/envelope.go
package aep

import "github.com/axisrobo/harmovela/event"

// ValidateEnvelope delegates structural validation to Event while retaining the
// legacy aep registry for non-Event domains.
func ValidateEnvelope(value map[string]any) []string {
	if value == nil {
		return event.ValidateEnvelope(value)
	}
	typ, isString := value["type"].(string)
	if !isString || event.IsStandardEventType(typ) {
		return event.ValidateEnvelope(value)
	}

	adapted := make(map[string]any, len(value))
	for key, item := range value {
		adapted[key] = item
	}
	adapted["type"] = "event.acknowledged"
	errors := event.ValidateEnvelope(adapted)
	if !IsStandardEventType(typ) {
		errors = append(errors, "type is not in the standard draft registry: "+typ)
	}
	return errors
}
